apmpackage/apm/data_stream/rum_traces/elasticsearch/ingest_pipeline/default.yml

```diff
@@ -38,4 +38,22 @@ processors:
         - service.framework
       ignore_missing: true
       ignore_failure: true
+  - script:
+      lang: painless
+      source: |
+        def representative_count = ctx?.span?.representative_count != null ? ctx?.span?.representative_count : ctx?.transaction?.representative_count;
+          
+        if(representative_count == null) {
+        return;
+        }
+        
+        int doc_count = (int)representative_count;
+        
+        double decimal = representative_count - doc_count;
+        
+        if (decimal > Math.random()) {
```

Comment:
it's distributing fractions of numbers based on the representative count, e.g. if the representative count is 2.4, there's a 60% chance the `_doc_count` will be 2, and 40% that it will be 3. This should help us get a better approximation of the underlying numbers. If we would always round, there could be a serious discrepancy.